management/src/main/java/org/kaazing/gateway/management/context/ManagementGatewayListener.java

```diff
@@ -69,8 +92,13 @@ public void startingService(ServiceContext serviceContext) {
     }
 
     @Override
+    public void stoppedService(ServiceContext serviceContext) {
+        serviceContext.getMonitoringFactory().close();
```

Comment:
We shouldn't be closing and freeing the counters when a service is stopped because it could be started again. This should be done only when the service is destroyed. All you need to do to fix this is rename this method to "destroyedService".